# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363926280 25200
#      Thu Mar 21 21:24:40 2013 -0700
# Node ID 022898b4323bebdc60633cc6432c893340b2a047
# Parent  fb27f1e082976df3b9b002cd9bd8c05b98f6727c
build: Disallow packaging distribution unless Java is present (bug #38532)

* scripts/Makefile.am: Add dist-hook which issues warning message and stops
'make dist' when Java is not present.

diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -433,16 +433,22 @@ install-pkg-add:
 .PHONY: install-pkg-add
 
 uninstall-pkg-add:
 	for f in $(PKG_ADD_FILES); do \
 	  rm -f $(DESTDIR)$(fcnfiledir)/$$f; \
 	done
 .PHONY: uninstall-pkg-add
 
+if AMCOND_HAVE_JAVA
+else
+dist-hook:
+	@echo "Packaging distribution requires Java." ; exit 1;
+endif
+
 EXTRA_DIST += \
   $(IMAGES) \
   $(FCN_FILES) \
   $(FCN_FILES_IN) \
   $(GEN_FCN_FILES) \
   DOCSTRINGS \
   mkdoc.pl \
   mk-pkg-add
